apps/gitness/src/pages-v2/repo/repo-summary.tsx

```diff
@@ -53,15 +62,38 @@ export default function RepoSummaryPage() {
 
   const { data: { body: repository } = {}, refetch: refetchRepo } = useFindRepositoryQuery({ repo_ref: repoRef })
 
+  const { data: { body: repoSummary } = {} } = useSummaryQuery({
+    repo_ref: repoRef,
+    queryParams: { include_commit: false, sort: 'date', order: 'asc', limit: 20, page: 1 }
+  })
+
+  const { branch_count, default_branch_commit_count, pull_req_summary, tag_count } = repoSummary || {}
+
   const { data: { body: branches } = {} } = useListBranchesQuery({
     repo_ref: repoRef,
-    queryParams: { include_commit: false, sort: 'date', order: 'asc', limit: 20, page: 1, query: '' }
+    queryParams: { include_commit: false, sort: 'date', order: orderSortDate.DESC, limit: branch_count }
```

Comment:
What if there were thousands of branches?

Reply:
yeah I've asked Hitesh to share if the api can support huge list of branches, otherwise will change it to query based approach.
the current implementation just unblocks us but is also similar to how github handles it.

Reply:
reverted this after discussion with Hitesh, will raise a new PR with search query hooked to the api